vpr/src/pack/re_cluster_util.h

```diff
@@ -107,7 +108,8 @@ bool pack_mol_in_existing_cluster(t_pack_molecule* molecule,
                                   std::unordered_set<AtomBlockId>* new_clb_atoms,
                                   bool during_packing,
                                   t_clustering_data& clustering_data,
-                                  t_lb_router_data*& router_data);
+                                  t_lb_router_data*& router_data,
+                                  const int& detailed_routing_stage = 1);
```

Comment:
Comment detailed_routing_stage parameter in the Doxygen block.
I think this would be better implemented as an int, not a const int &.

Reply:
I did it that way because I saw mode being passed as a const int& in one of the function.  Happy to change it.
detailed_routing_stage is an enum, not a bool; it might be more clear to #include the header that defines the enum so that the default value can be E_DETAILED_ROUTE_FOR_EACH_ATOM; I made this the default so that all the function calls in re_cluster.cpp don't need to be changed.

Reply:
We should make the variable type the enum type then, for clarity and to avoid potential (and subtle) possible later bugs if someone adds another enum value and the integer 1 is no longer the mode you want.  Including the header you need is fine.

Better to pass the enum by value than const ref, so that should be changed too.